custom_components/bms_ble/plugins/pro_bms.py

```diff
@@ -0,0 +1,543 @@
+        # Extract the raw value (lower 15 bits)
+        raw_value = int.from_bytes(current_bytes, byteorder="little") & 0x7FFF
+
+        # Check sign bit (bit 15)
+        if current_bytes[1] & 0x80:
+            # Negative current (discharging)
+            return -raw_value / 100.0
+        # Positive current (charging)
+        return raw_value / 100.0
+
+    def _validate_packet_length(self, packet_len: int) -> bool:
+        """Validate packet length."""
+        if packet_len != self._REALTIME_PACKET_LEN:
+            self._log.warning(
+                "Unexpected packet length: %d bytes (expected %d)",
+                packet_len,
+                self._REALTIME_PACKET_LEN
+            )
+            return False
+        return True
+
+    def _parse_fields_from_data(self, data_section: bytes) -> BMSsample:
+        """Parse fields from data section according to _FIELDS."""
+        result: BMSsample = {}
+        for key, offset, size, signed, func in self._FIELDS:
+            if offset + size <= len(data_section):
+                value = int.from_bytes(
+                    data_section[offset:offset + size],
+                    byteorder="little",
+                    signed=signed
+                )
+                result[key] = func(value)
+
+        return result
+
+    def _handle_current_and_charging(self, result: BMSsample, data_section: bytes) -> None:
+        """Handle battery charging status and protection status."""
+        if "current" not in result:
+            return
+
+        # Check discharge flag in byte 15 (bit 7)
+        byte15 = data_section[11]  # Byte 15 in full packet = offset 11 in data section
+        discharge_flag = (byte15 & 0x80) != 0
+
+        # Apply sign based on discharge flag
+        # Discharge flag set = discharging (negative current)
+        # Discharge flag clear = charging (positive current)
+        if discharge_flag:
+            result["current"] = -result["current"]
+
+        result["battery_charging"] = not discharge_flag
+
+        # Handle protection status from lower bits of byte 15
+        self._handle_protection_status(result, byte15 & 0x7F)
+
+    def _handle_protection_status(self, result: BMSsample, protection_byte: int) -> None:
+        """Check for protection/error status."""
+        if protection_byte:
+            result["problem_code"] = protection_byte
+            result["problem"] = True
+            # Log protection status
+            problems = []
+            for bit, problem in self._PROTECTION_BITS.items():
+                if protection_byte & bit:
+                    problems.append(problem)
+            # Always log if we have a protection byte
+            self._log.debug(
+                "Protection status detected: %s",
+                ", ".join(problems) if problems else "Unknown protection code"
+            )
+        else:
+            result["problem"] = False
+
+    def _handle_design_capacity(self, result: BMSsample) -> None:
```

Comment:
This function is not needed as there is no sensor for this value. Can be removed.